# Queries
Rerun last week's queries using bq command line tool (Paste your bq queries):     
What's the size of this dataset? (i.e., how many trips)

In [3]:
! bq query --use_legacy_sql=false 'SELECT count(*) FROM `bigquery-public-data.san_francisco.bikeshare_status`'

Waiting on bqjob_r46cd78ac0df26edf_000001689cae5fff_1 ... (0s) Current status: DONE   
+-----------+
|    f0_    |
+-----------+
| 107501619 |
+-----------+


What is the earliest start time and latest end time for a trip?

In [6]:
! bq query --use_legacy_sql=false 'SELECT min(start_date), max(end_date) FROM `bigquery-public-data.san_francisco.bikeshare_trips`'

Waiting on bqjob_r57ddeab8cbc293cf_000001689cb0935c_1 ... (0s) Current status: DONE   
+---------------------+---------------------+
|         f0_         |         f1_         |
+---------------------+---------------------+
| 2013-08-29 09:08:00 | 2016-08-31 23:48:00 |
+---------------------+---------------------+


How many bikes are there?

In [5]:
! bq query --use_legacy_sql=false 'SELECT COUNT(distinct bike_number) FROM `bigquery-public-data.san_francisco.bikeshare_trips`'

Waiting on bqjob_r61738775c3912939_000001689cb07025_1 ... (0s) Current status: DONE   
+-----+
| f0_ |
+-----+
| 700 |
+-----+


New Query (Paste your SQL query and answer the question in a sentence):
How many trips are in the morning vs in the afternoon?

In [10]:
! bq query --use_legacy_sql=false 'SELECT count(start_date) as morning_trips FROM `bigquery-public-data.san_francisco.bikeshare_trips` where TIME(start_date) < "12:00:00"'
! bq query --use_legacy_sql=false 'SELECT count(start_date) as afternoon_trips FROM `bigquery-public-data.san_francisco.bikeshare_trips` where TIME(start_date) > "12:00:00"'

Waiting on bqjob_r8f514030b8e1e1c_000001689cba392e_1 ... (0s) Current status: DONE   
+---------------+
| morning_trips |
+---------------+
|        412339 |
+---------------+
Waiting on bqjob_r7daf6422772b629_000001689cba497a_1 ... (0s) Current status: DONE   
+-----------------+
| afternoon_trips |
+-----------------+
|          570556 |
+-----------------+


There are 412,339 bike trips in the morning and 570,556 bike trips in the afternoon.

# Project Questions
Identify the main questions you'll need to answer to make recommendations (list below, add as many questions as you need).

Question 1: What end station is the most common for bikes to finish their trip?  How many?

Question 2: What starting station is the most common for bikes to begin their trip?  How many?

Question 3: Which zipcode is the most popular for bike sharing?  How many?

Question 4: What time/hour of the day do most bike trips start?

Question 5: What percentage of our bikes are being used in the morning?

# Answers
Answer at least 4 of the questions you identified above You can use either BigQuery or the bq command line tool. Paste your questions, queries and answers below.

Question 1: What end station is the most common for bikes to finish their trip?  How many?

Answer: San Francisco Caltrain (Townsend at 4th) - 92,014 trips

In [12]:
! bq query --use_legacy_sql=false 'SELECT count(end_station_name) as count, end_station_name FROM `bigquery-public-data.san_francisco.bikeshare_trips` group by end_station_name order by count desc'

Waiting on bqjob_r556f72706c54f455_000001689cd972ae_1 ... (0s) Current status: DONE   
+-------+-----------------------------------------------+
|  f0_  |               end_station_name                |
+-------+-----------------------------------------------+
| 92014 | San Francisco Caltrain (Townsend at 4th)      |
| 58713 | San Francisco Caltrain 2 (330 Townsend)       |
| 50185 | Harry Bridges Plaza (Ferry Building)          |
| 46197 | Embarcadero at Sansome                        |
| 44145 | 2nd at Townsend                               |
| 40956 | Market at Sansome                             |
| 39598 | Steuart at Market                             |
| 38545 | Townsend at 7th                               |
| 35477 | Temporary Transbay Terminal (Howard at Beale) |
| 26762 | Market at 4th                                 |
| 26746 | Powell Street BART                            |
| 25910 | Market at 10th                                |
| 23895 | 2nd at South Park                

Question 2: What starting station is the most common for bikes to begin their trip?  How many?

Answer: San Francisco Caltrain (Townsend at 4th) - 72,683 trips

In [11]:
! bq query --use_legacy_sql=false 'SELECT count(start_station_name) as count, start_station_name FROM `bigquery-public-data.san_francisco.bikeshare_trips` group by start_station_name order by count desc'

Waiting on bqjob_r3fcee7ba1e194d50_000001689cd94161_1 ... (0s) Current status: DONE   
+-------+-----------------------------------------------+
|  f0_  |              start_station_name               |
+-------+-----------------------------------------------+
| 72683 | San Francisco Caltrain (Townsend at 4th)      |
| 56100 | San Francisco Caltrain 2 (330 Townsend)       |
| 49062 | Harry Bridges Plaza (Ferry Building)          |
| 41137 | Embarcadero at Sansome                        |
| 39936 | 2nd at Townsend                               |
| 39200 | Temporary Transbay Terminal (Howard at Beale) |
| 38531 | Steuart at Market                             |
| 35142 | Market at Sansome                             |
| 34894 | Townsend at 7th                               |
| 30209 | Market at 10th                                |
| 27502 | Market at 4th                                 |
| 26218 | 2nd at South Park                             |
| 25204 | Powell Street BART               

Question 3: Which zipcode is the most popular for bike sharing?  How many?

Answer: 94107 - 106,913 trips

In [13]:
! bq query --use_legacy_sql=false 'SELECT zip_code, COUNT(*) as count FROM `bigquery-public-data.san_francisco.bikeshare_trips` group by zip_code order by count desc;'

Waiting on bqjob_r5d346b9a4eafc743_000001689cdaccb8_1 ... (0s) Current status: DONE   
+----------+--------+
| zip_code | count  |
+----------+--------+
| 94107    | 106913 |
| 94105    |  61232 |
| 94133    |  46544 |
| 94103    |  38072 |
| 94111    |  33642 |
| 94102    |  30222 |
| 94109    |  19043 |
| 95112    |  15420 |
| nil      |  15385 |
| 94158    |  13673 |
| 94611    |  13198 |
| 94117    |  12549 |
| 94403    |  10934 |
| 94602    |  10926 |
| 94610    |  10628 |
| 94110    |  10446 |
| 94040    |  10276 |
| 94501    |  10179 |
| 94114    |   9544 |
| 94025    |   9452 |
| 94010    |   9251 |
| 94108    |   9064 |
| 95110    |   8688 |
| 94070    |   8397 |
| 94041    |   7716 |
| 94608    |   7670 |
| 94404    |   7268 |
| 94087    |   7014 |
| 95113    |   6950 |
| 94402    |   6941 |
| 94002    |   6808 |
|          |   6803 |
| 94306    |   6709 |
| 94301    |   6493 |
| 94401    |   6473 |
| 94122    |   6295 |
| 94303    |   6145 |
| 94062    |   5749 |
| 94115    

Question 4: What time/hour of the day do most bike trips start?

Answer: 8 UTC - 132,464 trips
Note, this is an odd value since this would be 12am PST... Perhaps an 

In [14]:
! bq query --use_legacy_sql=false 'SELECT TIME_TRUNC(TIME(start_date), HOUR), count(TIME_TRUNC(TIME(start_date), HOUR)) as count FROM `bigquery-public-data.san_francisco.bikeshare_trips` group by TIME_TRUNC(TIME(start_date), HOUR) order by count desc;'

Waiting on bqjob_r3c9d095570050dbb_000001689ce95f28_1 ... (1s) Current status: DONE   
+----------+--------+
|   f0_    | count  |
+----------+--------+
| 08:00:00 | 132464 |
| 17:00:00 | 126302 |
| 09:00:00 |  96118 |
| 16:00:00 |  88755 |
| 18:00:00 |  84569 |
| 07:00:00 |  67531 |
| 15:00:00 |  47626 |
| 12:00:00 |  46950 |
| 13:00:00 |  43714 |
| 10:00:00 |  42782 |
| 19:00:00 |  41071 |
| 11:00:00 |  40407 |
| 14:00:00 |  37852 |
| 20:00:00 |  22747 |
| 06:00:00 |  20519 |
| 21:00:00 |  15258 |
| 22:00:00 |  10270 |
| 23:00:00 |   6195 |
| 05:00:00 |   5098 |
| 00:00:00 |   2929 |
| 01:00:00 |   1611 |
| 04:00:00 |   1398 |
| 02:00:00 |    877 |
| 03:00:00 |    605 |
+----------+--------+
